tests/unit/sagemaker/jumpstart/test_estimator.py

```diff
@@ -0,0 +1,463 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+from typing import Optional
+from unittest import mock
+import unittest
+
+import pytest
+
+from sagemaker.debugger.profiler_config import ProfilerConfig
+from sagemaker.instance_group import InstanceGroup
+
+from sagemaker.jumpstart.estimator import JumpStartEstimator
+
+from sagemaker.jumpstart.utils import get_jumpstart_content_bucket
+from tests.integ.sagemaker.jumpstart.utils import get_training_dataset_for_model_and_version
+from sagemaker.predictor import Predictor
+from sagemaker.session import Session
+from tests.unit.sagemaker.jumpstart.utils import (
+    get_special_model_spec,
+    overwrite_dictionary,
+)
+
```

Comment:
Could you please add test cases to ensure that:
- that the training image is retrieved with `TRAINING` scope
- that the hosting image is retrieved with `INFERENCE` scope on `deploy`
- if the user does not pass a training instance type, the default training instance is retrieved before retrieving the training image
- if the user passes a training instance type, the training image is retrieved with that instance
- that intelligent defaults for `vpc`, each `kms` key, `role` and `network_isolation` are correctly handled
- if the user does not pass a hosting instance type on `deploy`, the default hosting instance is retrieved before retrieving the hosting image
- if the user passes a hosting instance type, the hosting image is retrieved with that instance

If any of these tests are not possible at the top level, make sure you add coverage for them in the kwargs utils.